discovery/README.md

```diff
@@ -121,3 +121,40 @@ or incorrect metadata.
 The information obtained from service discovery is not considered sensitive
 security wise. Do not return secrets in metadata, anyone with access to
 the Prometheus server will be able to see them.
+
+
+## Writing an SD mechanism
+
+### The SD Interface
+
+A Service Discovery (SD) mechanism has to discover targets and provide them to Prometheus. It has to send the targets down to Prometheus in the form of [`[]TargetGroup`](https://godoc.org/github.com/prometheus/prometheus/config#TargetGroup).
+
+```
+- targets: ['10.11.150.1:7870', '10.11.150.4:7870']
```

Comment:
Hmm, it is actually the same thing: https://godoc.org/github.com/prometheus/prometheus/discovery#StaticProvider.Run